### PR TITLE
add table tutorial

### PR DESCRIPTION
**Pull request by melonora**
Added tutorial notebook for tables that deals with:
1. How to retrieve information on what elements a table is annotating.
2. How to change the annotation metadata of a table
3. How to construct tables that either annotate or do not annotate SpatialElements and how to add them to the SpatialData object.
4. How to perform SQL like joins on tables and SpatialElements in a SpatialData object.

### CODE REVIEW

File: notebooks/examples/attachments/joins_small.png
Comment (LucaMarconato) [inline notebooks/examples/attachments/joins_small.png:1]: nice! 😍

Comment (LucaMarconato) [reply notebooks/examples/attachments/joins_small.png:1]: just checking, is the image with the right license?

Comment (melonora) [reply notebooks/examples/attachments/joins_small.png:1]: I adjusted the original, ordering and naming is different

**Comment by LucaMarconato** [comment]
Thanks @melonora, great notebook! I extended it to add info on storing annotations within the `SpatialElement`s and getting the annotations with `get_values()`. Also I added some images and finally adjust the `.md` for the docs. Merging 🥳

**Pull request merged**